### PR TITLE
[Odie] Wire frontend with the backend

### PR DESCRIPTION
**Pull request by AllTerrainDeveloper**
Related to https://github.com/Automattic/ai-services/issues/555

## Proposed Changes

This patch is cleaning up a lot of code used for demo, in addition to connecting the front-end with the backend, loading existing chats from previous sessions or even clear the chat and start a new chat session

## Testing Instructions

We need to apply the patch D127022
Test the following flows:

**From search to message input filed.**
1. Open Help Center with the "wapuu" flag enabled.
2. Fill the search text box with anything you'd like
3. Click on Still need help?
4. The user input should be filled with the previous input
5. Go back
6. Assert that the previous search is now empty
7. Click again on Still need help?
8. Assert that the user input is empty

**On the Wapuu help center assistant**
1. Open Help Center
2. Click on Still Need Help button
3. Type anything and send the question
4. Wait for the response to arrive
5. Either close or reload the page
6. Go to the chat and assert that the chat is still present
7. Click on the three dots icon in the right top corner (next to the back button)
8. Click on "Start a new chat"
9. Assert that the conversation is cleared
10. Reload
11. Assert that the initial chat you had is no longer present

*

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by benzhu56 — APPROVED** [review]
Code LG, tested first section working great for me. How to access the second part? `On the Wapuu help center assistant`, is it the Wapuu popup in the bottom corner?

**Comment by AllTerrainDeveloper** [comment]
> Code LG, tested first section working great for me. How to access the second part? `On the Wapuu help center assistant`, is it the Wapuu popup in the bottom corner?

Oh! I should be more clear on the steps, what I mean is:

1. Open Help Center (top right ? icon)
2. Click on Still Need Help button
3. (Rest of the steps)

File: client/odie/context/get-odie-initial-message.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { preventWidows } from 'calypso/lib/formatting';
+import { Message, OdieAllowedBots } from '../types';
+
+const getOdieInitialPrompt = ( botNameSlug: OdieAllowedBots = 'wpcom-support-chat' ): string => {
+	switch ( botNameSlug ) {
+		case 'wpcom-support-chat':
+			return preventWidows(
+				'Hi there 👋 I’m Wapuu, WordPress.com’s AI assistant! Having an issue with your site or account? Tell me all about it and I’ll be happy to help.'
```
Comment (dereksmart) [inline client/odie/context/get-odie-initial-message.ts]: Do these need to be translated?

**Review by dereksmart — COMMENTED** [review]
The specific testing instructions are working as expected. The chats clear properly and the state of the plugin search fields are behaving well too. 

The only thing about the UX that I think could use improving is that my message (and loading indicator) gets buried after sending: 
https://github.com/Automattic/wp-calypso/assets/7129409/d3e56b6e-7778-4dbb-9973-c1394a7d09a6



It would also be great if after clicking "jump to recent", that it stayed at the recent, tracking and scrolling down along with it (so I don't have to keep clicking)

I have other observations and notes, though they are around the backend engine and responses themselves which we'll be addressing separately.

File: client/odie/context/get-odie-initial-message.ts (hunk 1 above)
Comment (AllTerrainDeveloper) [reply client/odie/context/get-odie-initial-message.ts]: I had the same concern, as we will launch only for the English language as the initial experiment... But you are right. We might forget even after the experiment.

**Comment by AllTerrainDeveloper** [comment]
> The specific testing instructions are working as expected. The chats clear properly and the state of the plugin search fields are behaving well too.
> 
> The only thing about the UX that I think could use improving is that my message (and loading indicator) gets buried after sending: https://github.com/Automattic/wp-calypso/assets/7129409/d3e56b6e-7778-4dbb-9973-c1394a7d09a6
> 
> It would also be great if after clicking "jump to recent", that it stayed at the recent, tracking and scrolling down along with it (so I don't have to keep clicking)
> 
> I have other observations and notes, though they are around the backend engine and responses themselves which we'll be addressing separately.

@dereksmart I will create an issue with everything you've suggested and found (except for the translation). On that ticket I also have some more work to do (collapsing Sources)

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9565027

Some locales (Hebrew) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @AllTerrainDeveloper, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Hi there 👋 I’m Wapuu, WordPress.com’s AI assistant! Having an issue with your site or account? Tell me all about it and I’ll be happy to help.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9565027).

**Pull request merged**